src/main/java/com/kodilla/ecommercee/controller/OrderController.java

```diff
@@ -1,77 +1,52 @@
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.Date;
 import java.util.List;
+import java.util.Optional;
 
 @RestController
 @RequestMapping("/v1/order")
 public class OrderController {
-    //lista wszystkich zamówień ok
-    //dodanie nowego zamówienia ok
-    //wyświetlenie zamówienia ok
-    //aktualizacja zamówienia ok
-    //usunięcie zamówienia ok
-
-    List<OrderDto> orders = Collections.emptyList();
+    @Autowired
+    private OrderRepository orderRepository;
+    @Autowired
+    private OrderMapper orderMapper;
 
     @GetMapping("/getOrders")
     private List<OrderDto> getOrders() {
-        return getOrderDtos();
+        return orderMapper.orderListToOrderDtoList(orderRepository.findAll());
     }
 
     @PostMapping("/createOrder")
     private void createOrder(@RequestBody OrderDto orderDto){
-        orders.add(orderDto);
+        orderRepository.save(orderMapper.orderDtoToOrder(orderDto));
     }
 
     @GetMapping("/getOrder")
     private OrderDto getOrder(@RequestParam("orderId") long orderId){
-        getOrderDtos();
-        for (OrderDto order:orders) {
-            if(order.getOrderId() == orderId){
-                return order;
-            }
-        }
-        return null;
+        Optional<Order> order = orderRepository.findById(orderId);
+        return orderMapper.orderToOrderDto(order.get());
     }
 
     @PutMapping("/updateOrder")
     private OrderDto updateOrder(@RequestBody OrderDto orderDto){
-        Long orderToUpdateId = orderDto.getOrderId();
-        List<ItemDto> updatedProducts = orderDto.getItems();
-        for (OrderDto theOrderDto:orders){
-            if(orderDto.getOrderId().equals(orderToUpdateId)){
-                theOrderDto.setItems(updatedProducts);
-            }
-        }
-        return orderDto;
+        orderRepository.save(orderMapper.orderDtoToOrder(orderDto));
+        Optional<Order> order = orderRepository.findById(orderDto.getOrderId());
+        return orderMapper.orderToOrderDto(order.get());
     }
 
     @DeleteMapping("/deleteOrder")
     public void deleteOrder(@RequestParam("id") long id){
-        getOrders();
-        for (int i = 0; i < orders.size(); i++) {
-            if (orders.get(i).getOrderId().equals(id)) {
-                orders.remove(i);
-            }
-        }
-    }
-
-    private List<OrderDto> getOrderDtos(){
-        User user = new User("TestName", false, new Cart());
-        List<ItemDto> productDtos1 = new ArrayList<>();
-        List<ItemDto> productDtos2 = new ArrayList<>();
-        OrderDto orderDto1 = new OrderDto(123L, new Date(), user,productDtos1, true);
-        OrderDto orderDto2 = new OrderDto(123L, new Date(), user,productDtos2, false);
-        orders.add(orderDto1);
-        orders.add(orderDto2);
-        return orders;
+        orderRepository.deleteById(id);
     }
```

Comment:
Nie powinniśmy chyba usuwać permanentnie rekordów z tabeli baz danych. Tutaj tak samo, co jak nie znajdzie ordera po id?

Reply:
tutaj jest ok. lepiej je wywalić. jeśli to zamówienie byłoby np w realizacji to wtedy fajnie mieć flagę `cancelled`, ale w takim przypadku jak tutaj to raczej bym wywalał